### PR TITLE
rpm:add kubeedge.spec for rpm build

### LINKED ISSUE #3155: rpm build support
yum is package management component in most operating system.
To use yum install kubeedge, we need rpm package first
So it is important to have a **rpm spec** file for user to build kubeedge rpm package~

### PR DESCRIPTION
**Pull request by CooperLi**
Signed-off-by: CooperLi <a710905118@163.com>

What type of PR is this?

/kind feature

What this PR does / why we need it:

add rpm spec file so that kubeedge can be compiled into rpm package and installed on os which has yum as package manager

Which issue(s) this PR fixes:

Fixes #3155

Special notes for your reviewer:
N/A

Does this PR introduce a user-facing change?:
N/A

### CODE REVIEW

**Comment by CooperLi** [comment]
The kubeedge.spec was tested on Fedora 34, CentOS8, openEuler 21.09

**Comment by subpathdev** [comment]
can you fix your commit message please?

**Comment by CooperLi** [comment]
> can you fix your commit message please?

@subpathdev I think it is ok now

**Review by fisherxu — COMMENTED** [review]
It's better to put all the rpm related files to `build` subdir, and create a new dir for them.

**Comment by fisherxu** [comment]
And could you please add some docs about how to use the kubeedge.spec?

File: kubeedge.spec [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+%{_unitdir}/edgecore.service
+%config(noreplace) %{_sysconfdir}/kubeedge/config/edgecore.example.yaml
+
+%files edgesite
+%license LICENSE
+%{_prefix}/local/bin/edgesite-agent
+%{_prefix}/local/bin/edgesite-server
+
+%changelog
+* Thu Sep 09 2021 lixiang<lixiang172@huawei.com> - 1.8.0-1
```
Comment (fisherxu) [inline kubeedge.spec:161]: The changelog is about is file?

Comment (CooperLi) [reply kubeedge.spec:161]: Yes, changelog is the log about spec file changes

**Comment by CooperLi** [comment]
> And could you please add some docs about how to use the kubeedge.spec?

sure，please take a look

**Review by fisherxu — COMMENTED** [review]
/approve

**Comment by gy95** [comment]
can open another PR to provide english materials 😄

**Comment by CooperLi** [comment]
> can open another PR to provide english materials 😄

OK~

**Review by fisherxu — COMMENTED** [review]
/lgtm

**Pull request merged**